hubverse_annotator/app.py

```diff
@@ -152,151 +352,132 @@ def load_hubverse_table(hub_file: UploadedFile | None):
-        observed_layers = target_data_chart(eh_to_plot)
-        forecast_and_observed_layers = forecast_layers + observed_layers
-        chart = (
-            forecast_and_observed_layers.facet(
-                row=alt.Row("model:N", title="Model"), columns=1
-            )
-        ).interactive()
-        st.altair_chart(chart, use_container_width=True)
-        # preference and comments saving
-        output_dir = pathlib.Path("../output")
-        output_dir.mkdir(parents=True, exist_ok=True)
-        annotations_file = output_dir / f"anno_{selected_ref_date}.json"
-        if annotations_file.exists():
-            with annotations_file.open("r") as f:
-                annotations = json.load(f)
-            logger.info(f"Annotations file created:\n{annotations_file}")
-        else:
-            annotations = {}
-        # save by location, with empty dict by default
-        by_loc_dict = annotations.setdefault(two_num_loc_abbr, {})
-        for model in selected_models:
-            st.markdown(f"### {model}")
-            # set status and comments keys, and get previous in json
-            status_key = f"status_{two_letter_loc_abbr}_{model}"
-            comment_key = f"comment_{two_letter_loc_abbr}_{model}"
-            prev = by_loc_dict.get(model, {})
-            default_status = prev.get("status", "None")
-            default_comment = prev.get("comment", "")
-            # select boxes for status and comments
-            status = st.selectbox(
-                "Status",
-                ["Preferred", "Omitted", "None"],
-                index=["Preferred", "Omitted", "None"].index(default_status),
-                key=status_key,
-            )
-            comment = st.text_input(
-                "Comments", value=default_comment, key=comment_key
-            )
-            # save in dictionary and write out
-            by_loc_dict[model] = {"status": status, "comment": comment}
-        with open(annotations_file, "w") as f:
-            json.dump(annotations, f, indent=2)
-
-        # export button
-        if st.button("Export forecasts"):
-            col1, col2, col3 = st.columns([1, 3, 1])
-            with col2:
-                st.success("Need export")
-
-    # record end time
-    end_time = time.time()
-    duration = end_time - start_time
-    logger.info(f"Session lasted around: {duration // 60} minutes.")
+
+    return smhubt_to_plot, eh_to_plot
+
+
+def main() -> None:
+    # record session start time
+    start_time = time.time()
+
+    # streamlit application begins
+    st.title("Forecast Annotator")
+
+    # super-mega hubverse forecast table required
+    eh_table, smhub_table = load_data()
+
+    if smhub_table.is_empty():
+        st.info("Please upload Hubverse Forecasts to begin.")
+        return
+
+    selected_ref_date, two_letter_loc_abbr, two_num_loc_abbr = (
+        render_ref_and_loc_controls(smhub_table)
+    )
+
+    smhubt_by_loc = smhub_table.filter(
```

Comment:
It's not just by loc, it's for the single selected loc.
```suggestion
    single_loc_smhubt = smhub_table.filter(
```